### DISCUSSION TITLE
HMR adds an extra slash to the file path

### DISCUSSION THREAD

**Discussion by ryinner** [webpack/webpack#16484] [Q&A] [answered]
Hey guys, I started vue 3 project with webpack-dev-serve. 
I try to use HMR, but webpack  adds an extra slash: `http://localhost:3000//app.87f68d4bf0a9dc08bb5a.hot-update.js` or misses domain, if i use ` publicPath: '/'`: `http://app.87f68d4bf0a9dc08bb5a.hot-update.js`.

Webpack config:
```JavaScript
const path = require("path");
const webpack = require("webpack");
const { VueLoaderPlugin } = require("vue-loader");

module.exports = {
    mode: "development",
    entry: {
        "/app": "./src/index.js",
    },
    output: {
        clean: true,
        path: path.resolve(__dirname, "public/assets/"),
        filename: "[name].js",
        chunkFilename: "[name].js",
        publicPath: '/'
    },
    stats: {
        errorDetails: true,
        chunks: false,
        assets: false,
        modules: false
    },
    module: {
        rules: [
            {
                test: /\.(css|scss|sass)$/i,
                use: [
                    "vue-style-loader",
                    "css-loader",
                    "sass-loader",
                ],
            },
            {
                test: /\.vue$/,
                loader: "vue-loader",
                options: {
                    loaders: {}
                }
            },
            {
                test: /\.js$/,
                loader: "babel-loader",
                exclude: /node_modules/
            },
            {
                test: /\.(png|jpg|gif|svg)$/,
                loader: "file-loader",
                options: {
                    name: "[name]-[hash].[ext]",
                    outputPath: "images"
                }
            }
        ]
    },
    resolve: {
        alias: {
            "@": path.resolve(__dirname, "src/"),
        },
        extensions: [".js", ".vue", ".json"]
    },
    devtool: "eval-cheap-source-map",
    plugins: [
        new VueLoaderPlugin(),
        new webpack.DefinePlugin({
            __VUE_OPTIONS_API__: false,
            __VUE_PROD_DEVTOOLS__: false
        }),
    ],
    devServer: {
        static:  path.join(__dirname, "public"),
        historyApiFallback: {
            index: "index.html"
        },
        hot: true,
        headers: {
            "Access-Control-Allow-Origin": "*"
        },
        host: 'localhost',
        compress: true,
        port: 3000,
    },
};
```
Packages:

```JSON
{
    "@testing-library/vue": "^6.6.1",
    "@vitejs/plugin-vue": "^3.2.0",
    "babel-loader": "^9.1.0",
    "css-loader": "^6.7.2",
    "dev-server": "^0.2.0",
    "eslint": "^8.27.0",
    "eslint-plugin-vue": "^9.7.0",
    "file-loader": "^6.2.0",
    "node-sass": "^8.0.0",
    "sass": "^1.56.1",
    "sass-loader": "^13.2.0",
    "style-loader": "^3.3.1",
    "vitest": "^0.25.2",
    "vue": "^3.2.45",
    "vue-router": "^4.1.6",
    "vue-loader": "^17.0.1",
    "vue-style-loader": "^4.1.3",
    "webpack": "^5.75.0",
    "webpack-cli": "^5.0.0",
    "webpack-dev-server": "^4.11.1"
}
```

System: Windows 10.

**Comment by alexander-akait** [comment, marked as answer]
Just remove `publicPath: '/'`, you don't need it for webpack v5 in most of cases

**Reply by ryinner** [reply]
I try without `publicPath`, but webpack  adds an extra slash between domain and file: `http://localhost:3000//app.87f68d4bf0a9dc08bb5a.hot-update.js`.

**Reply by alexander-akait** [reply]
How you inject scripts in your HTML file?

**Reply by ryinner** [reply]
```HTML
<script src="app.js"></script>
```

**Reply by alexander-akait** [reply]
I mean which a tool (plugin?) create this link, thank you

**Reply by ryinner** [reply]
I'm sorry, maybe I don't understand the question correctly. I run npm script `webpack serve --config path/to/config`. which watches for changes and after compilation gives the result along this path

**Reply by alexander-akait** [reply]
hm, I don't see HTML plugin in your configuration, how you generate HTML file? Also do you use the latest version of loaders and plugins (just check, because some bugs can be fixed)

**Reply by ryinner** [reply]
I use the the latest version of loaders. I inject script in static HTML file.

**Reply by alexander-akait** [reply]
>  I inject script in static HTML file.

Manually? I just need to figure out how and who generate `script` tags in your HTML, so we can undestand why we have problems with `/`

**Reply by ryinner** [reply]
Yes, I did it manually. Should I use some webpack plugin for generate html file?

**Reply by alexander-akait** [reply]
Yes, I recommend to try it, I think the problem somewhere with paths, anyway manual insertion should work too, can you create reproducible test repo and I will look

**Reply by ryinner** [reply]
I install html plugin into my project but that doent help. Error message: cannot get `http://127.0.0.1:3000//app.33d9cbf2800bade1847f.hot-update.js`

**Reply by alexander-akait** [reply]
Thank you for invite I will look at your repo soon

**Reply by alexander-akait** [reply]
@ryinner Sorry for delay, just remove `/` from entry, i.e. use:
```
entry: {
  "app": "./src/index.js",
},
```

Slash (`/`) at start used for another things (and allow to override publicPath and not only, but you don't need it)

**Reply by ryinner** [reply]
Thank you so much for your help.
